rest_server/SiodbLogger.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2019-2020 Siodb GmbH. All rights reserved.
+// Use of this source code is governed by a license that can be found
+// in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	CONSOLE = 1
+	FILE    = 2
+)
+
+const (
+	TRACE   = 1
+	DEBUG   = 2
+	INFO    = 3
+	WARNING = 4
+	ERROR   = 5
+	FATAL   = 6
+)
+
+type SiodbLogger struct {
+	mutex                    sync.Mutex
+	out                      io.Writer // logFile, os.Stdout
+	file                     *os.File
+	fileCreatedUnixTime      int64
+	channelType              uint
+	channelName              string
+	destination              string
+	maxLogFileSize           uint64
+	logFileExpirationTimeout uint64 // second
+	severityLevel            uint
+}
+
+func (logger *SiodbLogger) initLogger() (err error) {
+
+	// Derive log destination
+	switch strings.TrimSpace(strings.ToLower(logger.destination)) {
+	case "stdout":
+		logger.out = os.Stdout
+	case "stderr":
+		logger.out = os.Stderr
+	default: // Assuming it's a path
+		if err = logger.createNewLogFile(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (logger *SiodbLogger) closeLogFile() (err error) {
+	if err = logger.file.Close(); err != nil {
+		println(FormattedOutput(WARNING, "issue closing log file '%v': %v",
```

Comment:
print to stderr, not stdout

Reply:
println writes the result to standard error: https://golang.org/pkg/builtin/#print